src/poetry_plugin_pypi_proxy/utils.py

```diff
@@ -9,11 +25,54 @@ def get_repo_id(repo_url: str) -> str:
     Note that this can change between clones of this project.  This
     is used silently when caching packages and for the publisher, so
     it should have no effect on your build processes.
+
+    :param repo_url: The proxy repo url
+
+    :return repo id: A Unique Identifier for the proxy repo, this uses md5 hashing.
     """
     key = hashlib.md5((repo_url).encode()).hexdigest()
     return f"pypi-proxy-{key}"
 
 
 def parse_url(url: str) -> str:
-    """Remove suffix from url appropriately so it can be used by the proxy object."""
+    """
+    Remove suffix from url appropriately so it can be used by the proxy object.
+
+    :param url: A URL string for parsing
+
+    :return new_url: Returns the URL with simple suffix removed for using
+    in the proxy object
+
+    """
     return re.sub("/?(simple/?)?$", "", url)
+
+
+def generate_poetry_source_config(url: str) -> PoetrySourceConfig:
+    """
+    We need to account for a url with auth data to pass into Poetry with
+    the proxy URL
+
+    :param url: The URL string that we get from the env var
+
+    :raises ValueError: This error gets raised when we receive a
+    password without a username in the URL as this should not be possible.
+
+    :return PoetrySourceConfig: Returns a TypedDict that contains username,
+    password and a cleaned url
+
+    """
+    url_parts = urlparse(url)
+    cleaned_url = url_parts.scheme + "://" + url_parts.hostname + url_parts.path
```

Comment:
For this: `scheme` should be a check with `ValueError`, `hostname` may be always nonnull, `path` should be `(path or "")`.